File origsrc/libspatialite-5.0.1/config-msvc.h is a regular file while file src/libspatialite-5.0.1/config-msvc.h is a symbolic link
--- origsrc/libspatialite-5.0.1/configure.ac	2021-02-08 00:53:46.000000000 +0900
+++ src/libspatialite-5.0.1/configure.ac	2021-04-23 16:42:46.571242400 +0900
@@ -441,8 +441,7 @@ AC_ARG_ENABLE(minizip, [AS_HELP_STRING(
   [--enable-minizip], [enables MiniZIP inclusion [default=yes]])],
   [], [enable_minizip=yes])
 if test x"$enable_minizip" != "xno"; then
-  AC_CHECK_HEADERS(minizip/unzip.h,, [AC_MSG_ERROR([cannot find minizip/unzip.h, bailing out])])
-  AC_SEARCH_LIBS(unzLocateFile, minizip,,AC_MSG_ERROR(['libminizip' is required but it doesn't seem to be installed on this system.]))
+  PKG_CHECK_MODULES([MINIZIP], [minizip], , AC_MSG_ERROR(['libminizip' is required but it doesn't seem to be installed on this system.]))
   AC_DEFINE(ENABLE_MINIZIP)
 fi
 
--- origsrc/libspatialite-5.0.1/examples/Makefile.am	2021-02-08 00:53:46.000000000 +0900
+++ src/libspatialite-5.0.1/examples/Makefile.am	2021-04-23 16:43:13.142914500 +0900
@@ -6,7 +6,7 @@ noinst_PROGRAMS = demo1 demo2 demo3 demo
 endif
 
 AM_CFLAGS = -I@srcdir@/../src/headers 
-AM_LDFLAGS = -L../src -lspatialite -lm $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lm $(GCOV_FLAGS)
 
 LDADD = @LIBXML2_LIBS@
 
File origsrc/libspatialite-5.0.1/fakeconfig.h is a regular file while file src/libspatialite-5.0.1/fakeconfig.h is a symbolic link
--- origsrc/libspatialite-5.0.1/src/Makefile.am	2021-02-08 00:53:46.000000000 +0900
+++ src/libspatialite-5.0.1/src/Makefile.am	2021-04-23 20:44:02.103604500 +0900
@@ -45,17 +45,19 @@ libspatialite_la_LIBADD = ./gaiaaux/libg
 	./stored_procedures/libstored_procedures.la \
 	./connection_cache/libconnection_cache.la \
 	./virtualtext/libvirtualtext.la \
-	./wfs/libwfs.la @LIBXML2_LIBS@
+	./wfs/libwfs.la @LIBXML2_LIBS@ $(MINIZIP_LIBS)
+
+libspatialite_la_LDFLAGS = -no-undefined
 
 if MINGW
-libspatialite_la_LDFLAGS = -version-info 4:4:0 -no-undefined
+libspatialite_la_LDFLAGS += -version-info 4:4:0
 libspatialite_la_LIBADD += -lm
 else 
 if ANDROID
-libspatialite_la_LDFLAGS = -version-info 8:2:1
+libspatialite_la_LDFLAGS += -version-info 8:2:1
 libspatialite_la_LIBADD += -ldl -lm
 else
-libspatialite_la_LDFLAGS = -version-info 8:2:1
+libspatialite_la_LDFLAGS += -version-info 8:2:1
 libspatialite_la_LIBADD += -lpthread -ldl -lm
 endif
 endif
@@ -77,22 +79,21 @@ mod_spatialite_la_LIBADD = ./gaiaaux/gai
 	./stored_procedures/stored_procedures.la \
 	./connection_cache/connection_cache.la \
 	./virtualtext/virtualtext.la \
-	./wfs/wfs.la @LIBXML2_LIBS@
+	./wfs/wfs.la @LIBXML2_LIBS@ $(MINIZIP_LIBS)
 
 mod_spatialite_la_CPPFLAGS = @CFLAGS@ @CPPFLAGS@
 mod_spatialite_la_CPPFLAGS += -I$(top_srcdir)/src/headers -I.
 mod_spatialite_la_CPPFLAGS += -DLOADABLE_EXTENSION
 mod_spatialite_la_LIBTOOLFLAGS = --tag=disable-static
 
-if MINGW
 mod_spatialite_la_LDFLAGS = -module -avoid-version -no-undefined
+
+if MINGW
 mod_spatialite_la_LIBADD += -lm
 else 
 if ANDROID
-mod_spatialite_la_LDFLAGS = -module -version-info 8:0:1
 mod_spatialite_la_LIBADD += -ldl -lm
 else
-mod_spatialite_la_LDFLAGS = -module -version-info 8:0:1
 mod_spatialite_la_LIBADD += -lpthread -ldl -lm
 endif
 endif
File origsrc/libspatialite-5.0.1/src/headers/spatialite/gaiaconfig-msvc.h is a regular file while file src/libspatialite-5.0.1/src/headers/spatialite/gaiaconfig-msvc.h is a symbolic link
File origsrc/libspatialite-5.0.1/src/headers/spatialite/gaiaconfig.h is a regular file while file src/libspatialite-5.0.1/src/headers/spatialite/gaiaconfig.h is a symbolic link
--- origsrc/libspatialite-5.0.1/src/headers/spatialite.h	2021-02-08 00:53:46.000000000 +0900
+++ src/libspatialite-5.0.1/src/headers/spatialite.h	2021-04-23 16:45:53.054903800 +0900
@@ -48,7 +48,7 @@ the terms of any one of the MPL, the GPL
  Main SpatiaLite header file
  */
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 #ifdef DLL_EXPORT
 #define SPATIALITE_DECLARE __declspec(dllexport)
 #else
--- origsrc/libspatialite-5.0.1/src/shapefiles/shapefiles.c	2021-02-08 00:53:46.000000000 +0900
+++ src/libspatialite-5.0.1/src/shapefiles/shapefiles.c	2021-04-23 16:47:11.726284200 +0900
@@ -80,8 +80,12 @@ the terms of any one of the MPL, the GPL
 #endif
 
 #ifdef ENABLE_MINIZIP		/* MINIZIP is enabled */
+#if defined(__CYGWIN__)
+#include <unzip.h>
+#else
 #include <minizip/unzip.h>
 #endif
+#endif
 
 #if defined(_WIN32) && !defined(__MINGW32__)
 #define strcasecmp	_stricmp
--- origsrc/libspatialite-5.0.1/test/Makefile.am	2021-02-08 00:53:46.000000000 +0900
+++ src/libspatialite-5.0.1/test/Makefile.am	2021-04-23 16:47:54.120543800 +0900
@@ -114,9 +114,9 @@ endif
 AM_CFLAGS = -I@srcdir@/../src/headers -I@srcdir@
 
 if MINGW
-AM_LDFLAGS = -L../src -lspatialite -lm -lxml2 $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lm -lxml2 $(GCOV_FLAGS)
 else 
-AM_LDFLAGS = -L../src -lpthread -lspatialite -lm -lxml2 $(GCOV_FLAGS)
+AM_LDFLAGS = $(top_builddir)/src/libspatialite.la -lpthread -lm -lxml2 $(GCOV_FLAGS)
 endif
 
 TESTS = $(check_PROGRAMS)
